### ISSUE TITLE
Add support for `$id` as well as `id`

### ISSUE THREAD

**Issue by djmitche** [taskcluster/react-schema-viewer#23]
The current version only works with v6 schemas (with `schema.id`), but not v7 schemas (which have `schema.$id`).

**Comment by arshadkazmi42** [comment]
can you provide some more info for getting started with this bug. I would like to take a look into this.

**Comment by djmitche** [comment]
Sure -- sorry, I filed it quickly as a reminder to myself.

JSON-schema has a few [versions](http://json-schema.org/specification.html), and in Draft-06 a schema had an `id` property giving its id, while in Draft-07 this was renamed to `$id`.  

It's not practical to know, given a schema, which draft it corresponds to (its `$schema` property may point to some other, custom schema, [as we do in taskcluster](https://github.com/taskcluster/taskcluster-lib-references/blob/master/schemas/metadata-metaschema.yml)).  However, it shouldn't be too hard to always look for either `$id` or, if that's missing, for `id`.

So basically this bug involves changing references that look for `schema.id` to look first for `$id`.

**Comment by arshadkazmi42** [comment]
Thank you for the detailed info  @djmitche 
I am planning on doing a change something like this

```
schema.id -> schema.$id ? schema.$id : schema.id
```

**Comment by djmitche** [comment]
That sounds perfect.
